plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/StandardColumnMappings.java

```diff
@@ -501,7 +501,7 @@ public static LongReadFunction timestampReadFunction(TimestampType timestampType
 
     private static ObjectReadFunction longTimestampReadFunction(TimestampType timestampType)
     {
-        checkArgument(timestampType.getPrecision() > TimestampType.MAX_SHORT_PRECISION && timestampType.getPrecision() < MAX_LOCAL_DATE_TIME_PRECISION,
+        checkArgument(timestampType.getPrecision() > TimestampType.MAX_SHORT_PRECISION && timestampType.getPrecision() <= MAX_LOCAL_DATE_TIME_PRECISION,
```

Comment:
That's a correct fix, but it would be great to have a test, with some connector exercising this code path. Would be possible?

Reply:
Thanks for the feedback! Sure, let me develop a test